### PR TITLE
Fix tests and deprecations with Sidekiq 5

### PR DESCRIPTION
**Pull request by packrat386**
* Turn on BeDelayed with Sidekiq versions >= 5
* Test against old versions of Sidekiq
* Fix deprecation warning to only show on calling deprecated method(s)

### CODE REVIEW

**Comment by mlarraz** [comment]
This is not actually removed in Sidekiq 5, just disabled by default. See: https://github.com/mperham/sidekiq/blob/master/5.0-Upgrade.md

You just need to put something like

```ruby
Sidekiq::Extensions.enable_delay! if Sidekiq::VERSION > 4
```
in your test setup.

**Comment by trev** [comment]
@packrat386 

1. The BeDelayed matcher can still work in Sidekiq 5+ if enabled: https://github.com/mperham/sidekiq/blob/master/5.0-Upgrade.md#whats-new

I enable in order to test here: https://github.com/philostler/rspec-sidekiq/pull/126/files#diff-93830fa29d616f7c87903d08b5b1b29aR25

**Comment by trev** [comment]
Beat me to the punch @mlarraz!

**Comment by packrat386** [comment]
@trev cherry-picked your commits in here so I can make one green PR.

File: .travis.yml [hunk 1]
```diff
@@ -18,3 +18,5 @@ matrix:
   include:
     - rvm: 2.0
       gemfile: gemfiles/ruby20.gemfile
+    - rmv: 2.4
```
Comment (sshao) [inline .travis.yml]: should be `rvm`, i'm guessing

**Pull request merged**